Three branches to be merged in gnt

michelmno opened the pull request:
This work allow to trigger CI for  git push on top of gnt branch,

Here is summary of related commits, DETAILS IN COMMITS headers.

```
*-.   a5bca59a3 (HEAD -> tobemerged_in_gnt, origin/tobemerged_in_gnt) Three branches to be merged in gnt
|\ \  
| | * 2ceed5bc7 (origin/make_ci_gnt, make_ci_gnt) Comment out make ci in ci.yml as bypass issue #1793
| | * cfdd52a68 ci.yml from geneweb/master for gnt branch
| * | 98b7f546e (origin/move_calendars_back_gnt, move_calendars_back_gnt) camlp5 pin to 8.02.01 & ocamlformat to 0.24.1
| * | 979661ead Overwrite calendars.ml* from calendar-kind branch
| * | e5a5430da fmt changes for lib/util/calendars.ml*
| * | 245aa72e3 Move back calendars library from external
| |/  
* / 45d8669bb (origin/fix_odir_in_gnt, fix_odir_in_gnt) Fix management of -odir
|/  
*   dda4fc4dc (upstream/gnt, geneanet/gnt, gnt) Merge pull request #1811 from geneweb/gnt-gwc-nopub-opt
```

hgouraud committed:
Fix management of -odir

to be merged in gnt branch

This is cherry pick of cid ccd53b5 already merged in master branch
to solve issue #1334  (PR #1543)
===
commit ccd53b5f7196244e7988869579f25657587855cf
Author: Henri83 <henri.gouraud@laposte.net>
Date:   Sat Aug 5 19:03:51 2023 +0200

    Fix management of -odir
===

Plus two specific changes:
* remove new_data.gw ref that is useless as about
  old code trials before last commit ccd53b5.

* Use Files.mkdir_p not Mutil
  to avoid build error in gnt branch.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Move back calendars library from external

* initially moved out by cid 25b1407a
  commit 25b1407a80a5327f561a98ce39db679dfe55c35a (tag: Geneweb-25b1407a)
  Author: Julien Sagot <julien.sagot@geneanet.org>
  Date:   Thu Oct 31 07:59:29 2019 +0100
  Exported Calendar module as external lib

* I retrieved back inside because external library is not maintain in
  https://github.com/geneweb/calendars.git
  I retrieved the version 1.0.0 that is master branch cid 584fb84
  For tracking purpose here is the list of commits up to this version
  * 584fb84 Fixed dune version in opam file
  * 3382a99 Added first tests
  * 8841e50 Initialized project with code adapted from GeneWeb

* I only grabbed the two calendars.ml* files not the test/* file
  to avoid to adapt to new test engine.
  (subject to a potential futur commit if needed)

* Note that I voluntarily did not change source files
  even if badly formatted, to ease code comparison from calendars.git
  The next commit will do those format changes.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
fmt changes for lib/util/calendars.ml*

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Overwrite calendars.ml* from calendar-kind branch

* I retrieved the calendar-kind branch cid d1a36ad from
  https://github.com/geneweb/calendars.git
  For tracking purpose here is the list of commits since version 1.0.0
  (some concerns files that are not grabbed here)
  * d1a36ad fix issue with french complementary month
  * fa0b109 small clean up
  * 00c58f7 rename
  * 2a80ad9 test for gregorian too
  * c99364d better checks in make
  * 81c8711 remove unescessary conversion
  * 50a672e better error message
  * f7dee32 fix duplicate depends on dune
  * 43433c0 remove partial date conversion
  * 0406547 clean up; add comments
  * c456cb3 use gadt
  * 94f5503 change make to result
  * fb95512 add to_*
  * cf4069b add calendar kind
  * 8a6780e fmt
  * 557ba62 add .ocamlformat
  * a9c129b rename
  * c15b6d5 generate opam file
  * 3904b16 add license; readme

* Note that the branch "hebrew-0" is not in this commit
  is it required ?

* I only grabbed the two calendars.ml* files not the test/* file
  to avoid to adapt to new test engine.
  (subject to a potential futur commit if needed)

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
ci.yml from geneweb/master for gnt branch

* mandatory to allow  CI triggering for all pull requests
to gnt branch (in addition to PR to master branch)
* BUT limit Build & CI to ubuntu and ocaml 4.14.x
  (candidate for improvement in the futur)

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
camlp5 pin to 8.02.01 & ocamlformat to 0.24.1

* This is bypass of issue with camlp5 8.03.00
  https://github.com/camlp5/camlp5/issues/106

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno commented:
Note the failing make ci that is already tracked by issue #1793

ilankri commented:
@michelmno If your pull request is a work in progress (as suggested in its title), you can convert it to a draft (see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request).

michelmno committed:
Comment out make ci in ci.yml as bypass issue #1793

https://github.com/geneweb/geneweb/issues/1793
[BUG] make ci is failing on gnt branch

Signed-off-by: Michel Normand <michel_mno@laposte.net>

michelmno committed:
Three branches to be merged in gnt

michelmno commented:
close this ignored PR
